File: Day-6/Power_of_two.py
# FUNCTION
def Power_of_two(n):
    if(n==1):
        return True
    if(n<=0):
        return False
    while n%2==0:
        n=n//2
    if(n==1):
        return True
    return False
# Main Function
n=int(input("Input a number : "))
if(Power_of_two(n)):
    print("YES")
else : print("NO")
    

File: Day-6/Power_of_two.cpp
#include<iostream>
using namespace std;
bool power_of_two(int n)
{
    if(n==1)
    return true;
    if(n<=0)
    {
        return false;
    }
    while(n%2==0)
    {
        n/=2;
    }
    if(n==1)
    return true;
    return false;
}
int main()
{
    int n;
    cout<<"Input a number : ";
    cin>>n;
    if(power_of_two(n))
    {
        cout<<"YES";
    }
    else
    {
         cout<<"NO";
    }

}